### ISSUE TITLE
Error in specification example for array of connectors

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaSpecification#167]
**Modified by dietmarw on 28 Apr 2009 10:10 UTC**
Reported by Ralf Hannemann [Ralf.Hannemann@avt.rwth-aachen.de]:

The MatrixGain example in section 9.1 is incorrect (using size(A,1) for input-size). A corrected version of the example is:
```
 import Modelica.Blocks.Sources.Sine;
 connector InPort=input Real;
 connector OutPort=output Real;
 block MatrixGain
  input InPort u[size(A,2)];
  output OutPort y[size(A, 1)];
  parameter Real A[:,:]=[1];
 equation
    y = A*u;
 end MatrixGain;
 Sine sinSource[5];
 MatrixGain gain(A=5*identity(5));
 MatrixGain gain2(A=ones(2,5));
 OutPort x[2];
equation
 connect(sinSource.y, gain.u); // Legal
 connect(gain.y, gain2.u); // Legal
 connect(gain2.y, x); // Legal
```

The MatrixGain in MSL is correct.

----

**Reported by Ralf&#46;Hannemann on 28 Apr 2009 09:20 UTC**
Reported by Ralf Hannemann [Ralf.Hannemann@avt.rwth-aachen.de]:

The MatrixGain example in section 9.1 is incorrect (using size(A,1) for input-size). A corrected version of the example is:

 import Modelica.Blocks.Sources.Sine;
 connector InPort=input Real;
 connector OutPort=output Real;
 block MatrixGain
  input InPort u[size(A,2)];
  output OutPort y[size(A, 1)];
  parameter Real A[:,:]=[1];
 equation
    y = A\*u;
 end MatrixGain;
 Sine sinSource[5];
 MatrixGain gain(A=5\*identity(5));
 MatrixGain gain2(A=ones(2,5));
 OutPort x[2];
equation
 connect(sinSource.y, gain.u); // Legal
 connect(gain.y, gain2.u); // Legal
 connect(gain2.y, x); // Legal

The MatrixGain in MSL is correct.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/167

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 28 Apr 2009 10:10 UTC**
Put code in verbatim environment to avoid erroneous ticket links.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 28 Apr 2009 10:10 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 28 Apr 2009 10:11 UTC**
Added the real reporter to the reporter field.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 26 May 2009 23:32 UTC**
Fixed error in Modelica 3.1 specification.
